oneview/resource_hypervisor_manager.go

```diff
@@ -0,0 +1,395 @@
+			},
+
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"initial_scope_uris": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+		},
+	}
+}
+
+func resourceHypervisorManagerCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	hypMan := ov.HypervisorManager{
+		Name:        d.Get("name").(string),
+		DisplayName: d.Get("display_name").(string),
+		Username:    d.Get("username").(string),
+		Password:    d.Get("password").(string),
+		Port:        d.Get("port").(int),
+		Type:        d.Get("type").(string),
+	}
+
+	if val, ok := d.GetOk("initial_scope_uris"); ok {
+		rawInitialScopeUris := val.(*schema.Set).List()
+		initialScopeUris := make([]utils.Nstring, len(rawInitialScopeUris))
+		for i, raw := range rawInitialScopeUris {
+			initialScopeUris[i] = utils.Nstring(raw.(string))
+		}
+		hypMan.InitialScopeUris = initialScopeUris
+	}
+
+	preferencesList := d.Get("preferences").(*schema.Set).List()
+	for _, raw := range preferencesList {
+		preferences := raw.(map[string]interface{})
+		hypervisorManagerPreferences := ov.Preference{
+			Type:                     preferences["type"].(string),
+			VirtualSwitchType:        preferences["virtual_switch_type"].(string),
+			DistributedSwitchVersion: preferences["distributed_switch_version"].(string),
+			DistributedSwitchUsage:   preferences["distributed_switch_usage"].(string),
+			MultiNicVMotion:          preferences["multi_nic_v_motion"].(bool),
+			DrsEnabled:               preferences["drs_enabled"].(bool),
+			HaEnabled:                preferences["ha_enabled"].(bool),
+		}
+
+		hypMan.Preferences = &hypervisorManagerPreferences
+	}
+
+	resourcePathList := d.Get("resource_paths").(*schema.Set).List()
+	resourcePathCollect := make([]ov.ResourcePath, 0)
+	for _, raw := range resourcePathList {
+		resourcePaths := raw.(map[string]interface{})
+		hypervisorManagerResourcePaths := ov.ResourcePath{
+			UserPath:   resourcePaths["user_path"].(string),
+			ActualPath: resourcePaths["actual_path"].(string),
+		}
+		resourcePathCollect = append(resourcePathCollect, hypervisorManagerResourcePaths)
+		hypMan.ResourcePaths = resourcePathCollect
+	}
+	file, _ := json.MarshalIndent(hypMan, "", "")
```

Comment:
Remove these lines as this is for debugging purposes.

Reply:
DOne